### ISSUE TITLE
Machine Management

### ISSUE THREAD

**Issue by IvanPombo** [inductiva/inductiva#871]
**TLDR:** Machines are the bread and butter of computation and the most expensive part. Their management should be clear and intuitive.

Managing machines can be a pain... That is why Inductiva is also here! However, more awareness and safety needs to be brought to the users, so that it is clear that machines are up, they are in use, simulations are running in them, or they are non-used and can be deleted. 

The prospect of this issue is to gather the problems and propose solutions, to then split into several issues that can be more easily attacked. Here, we want to create the general overview.

### Problems:

- [ ] The management of machines should be more decoupled from the scripts used to launch simulations. With an interface to launch, terminate and visualise if machines are up, users can be more sure how many machines are available for them to use, launch and manage. The two clear interfaces for this purpose are the CLI and the web Console. They will still be able to launch in a script, but we can guide them to focus more on outer management.
- [ ] Merge Elastic and Normal Machine Groups class. This merge reduces duplicated code and classes which can be way less intuitive for users. Moreover, with the addition of MPI cluster and GPU capabilities, we can centralise everything in one class, and users can **attach** these capabilities as they wish.
- [ ] List of available machine types should come from the backend. Currently, it is just a list printed on the CLI.
- [ ] End machines after **x** time of no simulations being run. This is a long desired featured, since users may forget to kill their machines, and we should prevent that they get a large bill. This is more specific for Cloud usage.
